solidity/ecdsa/deploy/00_resolve_random_beacon_chaosnet.ts

```diff
@@ -0,0 +1,30 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+import type { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { deployments, helpers } = hre
+  const { log } = deployments
+
+  const RandomBeaconChaosnet = await deployments.getOrNull(
+    "RandomBeaconChaosnet"
+  )
+
+  if (
+    RandomBeaconChaosnet &&
+    helpers.address.isValid(RandomBeaconChaosnet.address)
+  ) {
+    log(
+      `using existing RandomBeaconChaosnet at ${RandomBeaconChaosnet.address}`
+    )
+  } else {
+    throw new Error("deployed RandomBeaconChaosnet contract not found")
+  }
+}
+
+export default func
+
+func.tags = ["RandomBeaconChaosnet"]
+
+// Only execute for chaosnet deployments.
+func.skip = async (hre: HardhatRuntimeEnvironment): Promise<boolean> =>
+  !hre.network.tags.chaosnet
```

Comment:
I think "chaosnet" tag will be a common point of confusion. Chaosnet is our rollout program (https://blog.threshold.network/tbtc-v2-hits-its-first-launch-milestone/) and the network we currently have deployed on mainnet is the Chaosnet. 

See that in `hardhat.config.ts`, in this PR, we tag `development` and `goerli` networks as Chaosnet and it is not the same as our `mainnet` Chaosnet program.

Here is what we are trying to do:
We have `RandomBeaconChaosnet` to deploy and to use it from `WalletRegistry` (proxy) contract.
- We do not want to start a revolution right now and rework a bunch of unit tests in `ecdsa/solidity` that work with a real `RandomBeacon`. So for `hardhat` network, we should use the real `RandomBeacon`. 
- I do not know about system tests (cc @dimpar) but I expect they use a real `RandomBeacon` as well given the `RandomBeaconChaosnet` was introduced quite recently.
- For Goerli, Mainnet, and local development (local geth) networks we want to use `RandomBeaconChaosnet`.

If we want to use the tag, then I would name it `randomBeaconChaosnet` to avoid confusion and tag three networks: `development`, `goerli`, and `mainnet`.

If we do not want to use the tag, I would specify here the networks explicitly. But keep in mind we'll have to modify `00_resolve_random_beacon.ts` as well, so maybe the tag works better?

Reply:
> If we want to use the tag, then I would name it `randomBeaconChaosnet` to avoid confusion and tag three networks: `development`, `goerli`, and `mainnet`.

Or maybe even `useRandomBeaconChaosnet` to be super-explicit about what is happening.